csharp/lang/correctness/sslcertificatetrust/sslcertificatetrust-handshake-no-trust.yaml

```diff
@@ -0,0 +1,23 @@
+rules:
+- id: correctness-sslcertificatetrust-handshake-no-trust
+  patterns:
+    - pattern-either:
+      - pattern: SslCertificateTrust.$METHOD($COLLECTION,sendTrustInHandshake=true)
+      - pattern: SslCertificateTrust.$METHOD($COLLECTION,true)
+    - metavariable-regex:
+        metavariable: $METHOD
+        regex: CreateForX509(Collection|Store)
+  fix: $METHOD($COLLECTION,false)
+  message: Sending the trusted CA list increases the size of the handshake request and can leak system configuration information.
+  languages: [csharp]
+  metadata:
+    references: 
+      - https://docs.microsoft.com/en-us/dotnet/api/system.net.security.sslcertificatetrust.createforx509collection?view=net-6.0#remarks
+      - https://docs.microsoft.com/en-us/dotnet/api/system.net.security.sslcertificatetrust.createforx509store?view=net-6.0#remarks
```

Comment:
Looking at both of these docs, it appears that both methods have exactly 2 args, is that the case? I just want to make sure that $COLLECTION will be sufficient, as if there are >2 potential args then it won't match some calls.

Reply:
Yes, these methods have exactly two args, though I suppose `$COLLECTION` might not capture an inline collection instantiation?